[Split Build] Test split build in CI

PaliC opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #127958
* __->__ #126699
* #126813
* #127934

Specifically this PR tests building the split build on our manywheel builds on trunk
 
Also until we come up with a better name "libtorch" was renamed to "libtorchsplit" so we don't accidentally download the pip package which screws up testing.

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

atalman commented on .github/workflows/generated-linux-aarch64-binary-manywheel-nightly.yml:
```diff
@@ -37,6 +37,70 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  manywheel-py3_8-cpu-aarch64-experimental-split-build-build:
```
Should we be targeting linux-aarch64 builds at this point ?

atalman commented on .github/workflows/generated-linux-s390x-binary-manywheel-nightly.yml:
```diff
@@ -37,6 +37,70 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
```
Please make sure s390x workflows are not affected

atalman commented on .github/workflows/generated-macos-arm64-binary-wheel-nightly.yml:
```diff
@@ -32,7 +32,7 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
```
Please make sure macos-arm64 workflows are not affected

atalman commented on .github/workflows/generated-windows-binary-wheel-nightly.yml:
```diff
@@ -32,7 +32,7 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
```
Please make sure windows workflows are not affected

PaliC committed:
Update on "[Split Build] Test split build in CI"


Specifically this PR tests building the split build on our manywheel builds on trunk
 
Also until we come up with a better name "libtorch" was renamed to "libtorchsplit" so we don't accidentally download the pip package which screws up testing.

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

huydhn commented on .ci/pytorch/build.sh:
```diff
@@ -280,20 +280,44 @@ else
         python -mpip install --pre numpy==2.0.0rc1
       fi
-      WERROR=1 python setup.py bdist_wheel
+      if [[ "$USE_SPLIT_BUILD" == true ]]; then
```
It looks like both this block and the one below are the same with only a difference in setting WERROR.  Let's try to see if we can combine them to avoid code duplication here

atalman commented on .ci/pytorch/build.sh:
```diff
@@ -333,7 +357,7 @@ else
     mkdir -p "$CUSTOM_OP_BUILD"
     pushd "$CUSTOM_OP_BUILD"
-    cmake "$CUSTOM_OP_TEST" -DCMAKE_PREFIX_PATH="$SITE_PACKAGES/torch" -DPYTHON_EXECUTABLE="$(which python)" \
+    cmake "$CUSTOM_OP_TEST" -DCMAKE_PREFIX_PATH="$SITE_PACKAGES/libtorchsplit;$SITE_PACKAGES/torch;$SITE_PACKAGES" -DPYTHON_EXECUTABLE="$(which python)" \
```
If we are not doing USE_SPLIT_BUILD here, just default path. Perhaps we should not pass ``$SITE_PACKAGES/libtorchsplit;`` in this case ?
We can do something like this:
```
SPLIT_PKG_PREFIX=""
if [[ "$USE_SPLIT_BUILD" == true ]]; then
SPLIT_PKG_PREFIX="$SITE_PACKAGES/libtorchsplit;"
fi
````
then include:
```
DCMAKE_PREFIX_PATH="$SPLIT_PKG_PREFIX$SITE_PACKAGES/torch;$SITE_PACKAGES" -DPYTHON_EXECUTABLE="$(which python)" 
```

PaliC replied:
ohh this is fair. Albeit if we are in a situation where this matters, we likely messed up.

But it's better to bug out early :)

ZainRizvi commented on .github/workflows/_binary-build-linux.yml:
```diff
@@ -16,6 +16,13 @@ on:
           default: linux.12xlarge
           type: string
           description: Hardware to run this "build"job on, linux.12xlarge or linux.arm64.2xlarge.
+      use_split_build:
```
What do you think about give the split build a more specific name in code?

E.g. "split_libtorch_build" or "use_libtorch_wheel"

atalman commented:
hi @PaliC main question is where can we observe the split results ? 
Looking the the nightly results, I see only 1 wheel being generated:
https://github.com/pytorch/pytorch/actions/runs/9209761460?pr=126699

Looking at the CI build and test, looks like it installs only 1 wheel before test:
https://github.com/pytorch/pytorch/actions/runs/9209759452/job/25335997850?pr=126699#step:19:130

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in CI"


Specifically this PR tests building the split build on our manywheel builds on trunk
 
Also until we come up with a better name "libtorch" was renamed to "libtorchsplit" so we don't accidentally download the pip package which screws up testing.

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

atalman commented:
@PaliC lets not apply this to [manywheel-py3_12-cpu-aarch64-build / build](https://hud.pytorch.org/pr/pytorch/pytorch/126699#25583463492) ([gh](https://github.com/pytorch/pytorch/actions/runs/9295861115/job/25583463492)) - for now. Size of binary 88mb, agin will be very small

atalman commented:
The main is that we are not doing split build, we still producing and installing only 1 package:
```
+ pip install /final_pkgs/torch-2.4.0.dev20240530+cu121-cp39-cp39-linux_x86_64.whl --index-url https://download.pytorch.org/whl/nightly/cu121
Looking in indexes: https://download.pytorch.org/whl/nightly/cu121
Processing /final_pkgs/torch-2.4.0.dev20240530+cu121-cp39-cp39-linux_x86_64.whl
Collecting filelock (from torch==2.4.0.dev20240530+cu121)
  Downloading https://download.pytorch.org/whl/nightly/filelock-3.13.1-py3-none-any.whl (11 kB)
Collecting typing-extensions>=4.8.0 (from torch==2.4.0.dev20240530+cu121)
  Downloading https://download.pytorch.org/whl/nightly/typing_extensions-4.8.0-py3-none-any.whl (31 kB)
Collecting sympy (from torch==2.4.0.dev20240530+cu121)
  Downloading https://download.pytorch.org/whl/nightly/sympy-1.12-py3-none-any.whl (5.7 MB)
25l     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0.0/5.7 MB ? eta -:--:--
     ━━━━━━━━━━━━━━━━━━━━━╺━━━━━━━━━━━━━━━━━━ 3.1/5.7 MB 92.4 MB/s eta 0:00:01
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╸ 5.7/5.7 MB 90.9 MB/s eta 0:00:01
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.7/5.7 MB 67.2 MB/s eta 0:00:```

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in CI"


Specifically this PR tests building the split build on our manywheel builds on trunk
 
Also until we come up with a better name "libtorch" was renamed to "libtorchsplit" so we don't accidentally download the pip package which screws up testing.

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in CI"


Specifically this PR tests building the split build on our manywheel builds on trunk
 
Also until we come up with a better name "libtorch" was renamed to "libtorchsplit" so we don't accidentally download the pip package which screws up testing.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in CI"


Specifically this PR tests building the split build on our manywheel builds on trunk
 
Also until we come up with a better name "libtorch" was renamed to "libtorchsplit" so we don't accidentally download the pip package which screws up testing.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in CI"


Specifically this PR tests building the split build on our manywheel builds on trunk
 
Also until we come up with a better name "libtorch" was renamed to "libtorchsplit" so we don't accidentally download the pip package which screws up testing.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in CI"


Specifically this PR tests building the split build on our manywheel builds on trunk
 
Also until we come up with a better name "libtorch" was renamed to "libtorchsplit" so we don't accidentally download the pip package which screws up testing.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in CI"


Specifically this PR tests building the split build on our manywheel builds on trunk
 
Also until we come up with a better name "libtorch" was renamed to "libtorchsplit" so we don't accidentally download the pip package which screws up testing.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in CI"


Specifically this PR tests building the split build on our manywheel builds on trunk
 
Also until we come up with a better name "libtorch" was renamed to "libtorchsplit" so we don't accidentally download the pip package which screws up testing.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in CI"


Specifically this PR tests building the split build on our manywheel builds on trunk
 
Also until we come up with a better name "libtorch" was renamed to "libtorchsplit" so we don't accidentally download the pip package which screws up testing.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in CI"


Specifically this PR tests building the split build on our manywheel builds on trunk
 
Also until we come up with a better name "libtorch" was renamed to "libtorchsplit" so we don't accidentally download the pip package which screws up testing.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in CI"


Specifically this PR tests building the split build on our manywheel builds on trunk
 
Also until we come up with a better name "libtorch" was renamed to "libtorchsplit" so we don't accidentally download the pip package which screws up testing.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in CI"


Specifically this PR tests building the split build on our manywheel builds on trunk
 
Also until we come up with a better name "libtorch" was renamed to "libtorchsplit" so we don't accidentally download the pip package which screws up testing.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in CI"


Specifically this PR tests building the split build on our manywheel builds on trunk
 
Also until we come up with a better name "libtorch" was renamed to "libtorchsplit" so we don't accidentally download the pip package which screws up testing.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in CI"


Specifically this PR tests building the split build on our manywheel builds on trunk
 
Also until we come up with a better name "libtorch" was renamed to "libtorchsplit" so we don't accidentally download the pip package which screws up testing.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in CI"


Specifically this PR tests building the split build on our manywheel builds on trunk
 
Also until we come up with a better name "libtorch" was renamed to "libtorchsplit" so we don't accidentally download the pip package which screws up testing.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in CI"


Specifically this PR tests building the split build on our manywheel builds on trunk
 
Also until we come up with a better name "libtorch" was renamed to "libtorchsplit" so we don't accidentally download the pip package which screws up testing.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in CI"


Specifically this PR tests building the split build on our manywheel builds on trunk
 
Also until we come up with a better name "libtorch" was renamed to "libtorchsplit" so we don't accidentally download the pip package which screws up testing.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in CI"


Specifically this PR tests building the split build on our manywheel builds on trunk
 
Also until we come up with a better name "libtorch" was renamed to "libtorchsplit" so we don't accidentally download the pip package which screws up testing.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in CI"


Specifically this PR tests building the split build on our manywheel builds on trunk
 
Also until we come up with a better name "libtorch" was renamed to "libtorchsplit" so we don't accidentally download the pip package which screws up testing.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in CI"


Specifically this PR tests building the split build on our manywheel builds on trunk
 
Also until we come up with a better name "libtorch" was renamed to "libtorchsplit" so we don't accidentally download the pip package which screws up testing.

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Test split build in CI"


Specifically this PR tests building the split build on our manywheel builds on trunk
 
Also until we come up with a better name "libtorch" was renamed to "libtorchsplit" so we don't accidentally download the pip package which screws up testing.

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]
